### PR TITLE
Add support for SFTP SSH key authentication

### LINKED ISSUE #5: SSH Key
I can't find an example of how to use sftp with a key (in my case without password) instead of a password, is this supported right now?

### PR DESCRIPTION
**Pull request by martinmaillard**
Adds `privateKey` and `passphrase` options to SFTP client. `privateKey` should be the path to the private key file.

This PR also changes the default SFTP port (21 -> 22) and fixes authentication in the test SFTP server.

Fixes #5

### CODE REVIEW

**Review by bergos — APPROVED** [review]

File: lib/SftpClient.js [hunk 1]
```diff
@@ -1,24 +1,31 @@
...
   async connect () {
-    this.session = await this.client.session({
+    const options = {
       host: this.host,
       port: this.port,
       username: this.user,
-      password: this.password
-    })
+      password: this.password,
+      privateKey: this.privateKey && await fs.readFile(this.privateKey),
```
Comment (bergos) [inline lib/SftpClient.js]: I think this should be a string and the caller has to take care of how to get the key string. This is useful for cases where the pipeline runs in a GitLab instance and the key is provides as env variable. Reading from the file system is still possible, but right now a little bit complicated: 

```
"${require('fs').readFileSync('/full/path/myKeyFile.key').toString()}"^^code:EcmaScriptTemplateLiteral
```
We should add a loader to read the content of files on the local file system via datatype.

Comment (ktk) [reply lib/SftpClient.js]: I agree getting this from ENV is more likely in a pipeline. I do that already in my case.

Comment (martinmaillard) [reply lib/SftpClient.js]: Ok, will change

**Comment by martinmaillard** [comment]
The `privateKey` param now takes the actual key content. Users have to take care of loading it from file.

I will merge this and solve the read issue in a separate PR. It's not really related.

**Pull request merged**